storcli.py

```diff
@@ -57,8 +64,178 @@ def main(args):
     except KeyError:
         pass
 
+    parsers = {
+        'cv': handle_cachevault,
+        'bbu': handle_bbu
+    }
+
+    for parser, parser_func in parsers.items():
+        data = get_storcli_json('/cALL/' + parser + ' show all J')
+
+        try:
+            data = data['Controllers']
+            for controller in data:
+                if controller['Command Status']['Status'] in ['Success']:
+                    #print('# Executing parser: ' + parser + ' = ' + str(parser_func) + ': ' + str(controller))
+                    response = controller['Response Data']
+                    controller_index = controller['Command Status']['Controller']
+                    parser_func(args, controller_index, response)
+        except KeyError:
+            pass
+
     print_all_metrics(metric_list)
 
+
+# We don't expect BBUs/CVs bigger than GB nor smaller than MB
+def get_amount_in_megabytes(number, units):
+    if units in ['GB']:
+        return float(number) * 1024
+    return number
+
+def search_property(response_data, section, property):
+    for pair_kv in response_data[section]:
+        if property == pair_kv['Property']:
+            return str(pair_kv['Value']).strip()
+    return ''
+
+# The following is not really TZ aware, deal with caution :-)
```

Comment:
To avoid the TZ problem: take the difference between the learn timestamp and the controller's current clock time, and add this to the system utime.